src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,14 +42,56 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        
+        final String[] splitAddressComponents = address.trim().split(",");
+        
+    	this.block = new Block(splitAddressComponents[0].trim());
+    	this.street = new Street(splitAddressComponents[1].trim());
+    	this.unit = new Unit(splitAddressComponents[2].trim());
+    	this.postalCode = new PostalCode(splitAddressComponents[3].trim());
+    	this.value = address;
     }
-
+    
     /**
-     * Returns true if a given string is a valid person email.
+     * Returns true if a given string is a valid person address.
      */
     public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+        if(!test.matches(ADDRESS_VALIDATION_REGEX))
+        	return false;
+        
+        final String[] splitAddressComponents = test.trim().split(",");
+        if(!isValidAddress(splitAddressComponents))
+        	return false;
```

Comment:
This doesn't follow our coding convention, please check them out. You need to put the brackets even for one-line code.